enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,616 @@
+
+%% Decisions
+
+  data-dir-exists?{/var/lib/microshift/}
+  backup-or-restore?{Backup or restore\nbased on previous boot}
+  backup-exists?{Does backup exist?}
+  version?{Compare\nMicroShift's version:\nmetadata vs binary}
+  data-migration?{Was\nmigration\nsuccessful?}
+  healthy?{"Is MicroShift healthy?\n(greenboot check)"}
+  missing-metadata-version?{Upgrade\nfrom 4.13 is\nsupported}
+  restore-ok?{Restore\nsucceeded?}
+  backup-ok?{Backup\nsucceeded?}
+
+%% Transitions
+
+  start --> pre-run --> data-dir-exists?
+
+  data-dir-exists? -- Doesn't exists --> run
+  data-dir-exists? -- Exists         --> backup-or-restore?
+
+  backup-or-restore? -- "Restore (red)"    --> do-restore
+  backup-or-restore? -- "Backup (green)"   --> do-backup
+  backup-or-restore? -- Missing\nData exists,\nbut no info persisted about what to do\nFits upgrade from 4.13 flow --> do-backup
+
+  do-backup --> backup-ok?
+  backup-ok? -- Yes --> version?
+  backup-ok? -. No .-> exit
+
+  do-restore                             --> backup-exists?
+  backup-exists? -- No --> do-clean-data --> run
+  backup-exists? -- Yes --> restore-ok? 
+  restore-ok? -- Yes --> version?
+  restore-ok? -. No .-> exit
+
+  version? -- Versions are the same                 --> run
+  version? -. Binary is older                       .-> exit
+  version? -- Binary is newer                       --> do-data-migration
+  version? -. Upgrade is blocked                    .-> exit
+  version? -- Upgrade is allowed                    --> do-data-migration
+  version? -- "Metadata is missing\n(assume 4.13)"  --> missing-metadata-version?
+
+  missing-metadata-version? -- Yes --> do-data-migration
+  missing-metadata-version? -. No .-> exit
+
+  do-data-migration --> data-migration?
+  data-migration? -- Yes --> run
+  data-migration? -. No  .-> exit
+
+  run --> healthy?
+  healthy? -- Yes --> green -->  continue-running
+  healthy? -- Yes --> do-persist-version
+  healthy? -. No  .-> red
+
+  exit -.-> red
+  red --> reboot
+```
+
+### API Extensions
+
+Metadata persisted on filesystem related to the functionality described 
+in this enhancement such as version metadata, next boot action, and any other,
+is considered internal implementation detail and not an API to be consumed.
+Content (schema) and location of these files are subject to change.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+Being a GA feature from the beginning the risks are not foreseeing fail scenarios in advance and implementation bugs
+that are not caught and fixed through graduation process.
+
+To mitigate the risks, a thorough review of the enhancement must be done by MicroShift, OpenShift, and RHEL teams,
+and making sure testing strategy is sound and prioritized equally with the feature development.
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+### Definitions
+
+- **ostree commit**: TODO
+- **ostree deployment**: TODO
```

Comment:
Does this mean the current ostree commit?
`rpm-ostree status -b`

Reply:
I haven't yet decided which term is more accurate in this context, see:
- https://ostreedev.github.io/ostree/repo/#commit-objects
- https://ostreedev.github.io/ostree/deployment/